assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+                    allTeams.add(new Team(temp, play));
+                }
+            }
+
+            br.close();
+        } catch (FileNotFoundException fnfe) {
+            System.out.println(fnfe.getMessage());
+        } catch (IOException ioe) {
+            System.out.println(ioe.getMessage());
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    //Second Task
+    public static void matchFixture() throws FileNotFoundException {
+        try {
+            PrintWriter pw = new PrintWriter(new File("fixture.csv"));
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("Sr.No");
+            sb.append(",");
+            sb.append("Guest Team");
+            sb.append(",");
+            sb.append(("Home Team"));
+            sb.append(",");
+            sb.append("Venue-Ground");
+            sb.append("\n");
+            Set<String> hash_Set = new HashSet<String>();
+            for (Players t : allPlayers) {
+                hash_Set.add(t.getTeam());
+            }
+            Integer srNo = 0;
+            for (String s : hash_Set) {
+                for (String t : hash_Set) {
+                    if (s != t) {
+                        srNo++;
+//                        System.out.println(s + " " + t);
+                        sb.append(srNo);
+                        sb.append(",");
+                        sb.append(s);
+                        sb.append(",");
+                        sb.append(t);
+                        sb.append(",");
+                        sb.append("null");
+                        sb.append("\n");
+                    }
+                }
+            }
+
+            pw.write(sb.toString());
+            pw.close();
+            System.out.println("fixture.csv file created!");
+
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+
+    }
+
+//    Third Task
+    public static void bowlersWithLeast40Wickets(String bowlersTeam) {
+        try {
+            for (Players p : allPlayers) {
+                if (p.getTeam().equals(bowlersTeam)) {
+                    if (p.getWickets() >= 40) {
+                        System.out.println(p.getName() + " " + p.getWickets());
+                    }
+                }
+            }
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    //Fourth Task
+    public static void searchPlayer(String playerName) {
+        try {
+            boolean flag = true;
+            for(Players p : allPlayers) {
+                if(p.getName().contains(playerName))
+                {
+                    flag = true;
+                    System.out.println("Name: " + p.getName());
+                    System.out.println("Team: " + p.getTeam());
+                    System.out.println("Role: " + p.getRole());
+                    System.out.println();
+                }
+                else {
```

Comment:
This isn't the correct implementation
consider the situation where the last element of the players list did not contain the keyword, then the flag would be set to false, hence your method would print the list of players and also print that players were not found